Android build fail when if productFlavors configurated

### Steps to Reproduce the Problem

Add `productFlavors` prod & staging

try to clean build/assemble apk
```
./android/gradlew clear -p android
```

it will fail `Cannot invoke method finalizedBy() on null object`


**more details:** 

```java
// instabug-reactnative/android/sourcemaps.gradle

    def bundleTask = appProject.tasks.find {
        task -> task.name.endsWithIgnoreCase('bundleReleaseJsAndAssets')
    }

   bundleTask // null - 
   // ^^^ because when android has flavors task will be called:
   // - bundleProdReleaseJsAndAssets
   // - bundleStagingReleaseJsAndAssets

    bundleTask.finalizedBy uploadSourcemaps
```

### Expected Behavior

no errors

### Actual Behavior

```
FAILURE: Build failed with an exception.

* Where:
Script 'node_modules/instabug-reactnative/android/sourcemaps.gradle' line: 11

* What went wrong:
Cannot invoke method finalizedBy() on null object

```

### Instabug integration code

no

### SDK Version

the latest

### React Native, iOS and Android Versions

android

### Device Model



### [Optional] Project That Reproduces the Issue

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,7 @@
 ### Fixed
 
 - Fix an issue with the Android sourcemaps upload script, causing the build to fail on older versions of Gradle ([#970](https://github.com/Instabug/Instabug-React-Native/pull/970)), closes [#969](https://github.com/Instabug/Instabug-React-Native/issues/969).
+- Fix an issue with the Android sourcemaps upload script, causing the build to fail when using product flavors ([#975](https://github.com/Instabug/Instabug-React-Native/pull/975)), closes [#974](https://github.com/Instabug/Instabug-React-Native/issues/974).
 
 ## [11.10.0](https://github.com/Instabug/Instabug-React-Native/compare/v11.9.1...11.10.0) (April 20, 2023)
 

--- a/android/sourcemaps.gradle
+++ b/android/sourcemaps.gradle
@@ -5,10 +5,7 @@ def appProject = project(":app")
 gradle.projectsEvaluated {
     // Works for both `bundleReleaseJsAndAssets` and `createBundleReleaseJsAndAssets` and product flavors
     def bundleTask = appProject.tasks.find {
-        def task = it.name
-        def hasPrefix = task.startsWith('bundle') || task.startsWith('createBundle')
-        def hasSuffix = task.endsWith('ReleaseJsAndAssets')
-        return hasPrefix && hasSuffix
+        task -> task.name.endsWith('ReleaseJsAndAssets')
     }
 
     bundleTask.finalizedBy uploadSourcemaps

--- a/android/sourcemaps.gradle
+++ b/android/sourcemaps.gradle
@@ -3,9 +3,12 @@ import org.apache.tools.ant.taskdefs.condition.Os
 def appProject = project(":app")
 
 gradle.projectsEvaluated {
-    // Works for both `bundleReleaseJsAndAssets` and `createBundleReleaseJsAndAssets`
+    // Works for both `bundleReleaseJsAndAssets` and `createBundleReleaseJsAndAssets` and product flavors
     def bundleTask = appProject.tasks.find {
-        task -> task.name.toLowerCase().endsWith('bundlereleasejsandassets')
+        def task = it.name
+        def hasPrefix = task.startsWith('bundle') || task.startsWith('createBundle')
+        def hasSuffix = task.endsWith('ReleaseJsAndAssets')
+        return hasPrefix && hasSuffix
     }
 
     bundleTask.finalizedBy uploadSourcemaps

